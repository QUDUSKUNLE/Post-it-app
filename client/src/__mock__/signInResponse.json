{
    "message": "User Signed in successfully",
    "response": [{
        "-KwW2TEuvqwcF2ODZH1-": {
            "phone_Number": "07031187445",
            "time": "Sun, Oct 15, 2017 6:21 PM",
            "userEmail": "quduskunle@gmail.com",
            "userId": "NCaAzr0ZzqfCLtXQlQG0jW2DWbg1",
            "userName": "kunle"
        }
    }]
}