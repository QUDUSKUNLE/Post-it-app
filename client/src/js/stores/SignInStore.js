import { EventEmitter } from 'events';
import AppDispatcher from '../dispatcher/AppDispatcher.js';
import {
  SIGN_IN_SUCCESS, SIGN_IN_ERROR
} from '../constants/ActionConstants.js';

/**
 * Signout Store, it hold user's state, listen to signin Actions
 * @class SignInStore
 */
class SignInStore extends EventEmitter {
  constructor(props) {
    super(props);
    this.handleActions = this.handleActions.bind(this);
  }

  handleActions(action) {
    switch (action.type) {
      case SIGN_IN_SUCCESS:
        this.emit('SIGN_OUT_SUCCESS');
        break;

      case SIGN_IN_ERROR:
        this.emit('SIGN_IN_ERROR');
        break;

      default:
    }
  }
}

// Initiate an instance of SignInStore
const signInStore = new SignInStore();

// Register a dispatcher and bind it to handleActions method
AppDispatcher.register(signInStore.handleActions.bind(signInStore));

// export an instance of SignInStore
export default signInStore;
