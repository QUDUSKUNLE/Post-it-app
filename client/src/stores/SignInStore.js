import { EventEmitter } from 'events';
import AppDispatcher from '../dispatcher/AppDispatcher';
import {
  SIGN_IN_SUCCESS,
  GOOGLE_SIGN_IN_SUCCESS, PASSWORD_RESET_SUCCESS
} from '../constants/ActionConstants';

/**
 * Signout Store, it hold user's state, listen to signin Actions
 * @class SignInStore
 */
class SignInStore extends EventEmitter {
  /**
   * @constructor
   */
  constructor() {
    super();
    this.signInMessage = {};
    this.googleSignInMessage = {};
    this.passwordResetResponse = {};
    this.signInUser = this.signInUser.bind(this);
    this.googleSignIn = this.googleSignIn.bind(this);
    this.passwordReset = this.passwordReset.bind(this);
    this.handleActions = this.handleActions.bind(this);
  }

  signInUser() {
    return this.signInMessage;
  }

  googleSignIn() {
    return this.googleSignInMessage;
  }

  passwordReset() {
    return this.passwordResetResponse;
  }

  /**
   * Receives actions and update the stores accordingly
   * @method handleActions
   * @param {object} action - Action type and data
   * @return {*} -
   */
  handleActions(action) {
    switch (action.type) {
      case SIGN_IN_SUCCESS:
        this.signInMessage = action.response;
        this.emit('SIGN_IN_SUCCESS');
        break;

      case GOOGLE_SIGN_IN_SUCCESS:
        this.googleSignInMessage = action.response;
        this.emit('GOOGLE_SIGN_IN_SUCCESS');
        break;

      case PASSWORD_RESET_SUCCESS:
        this.passwordResetResponse = action.response;
        this.emit('PASSWORD_RESET_SUCCESS');
        break;
      default:
    }
  }
}

// Initiate an instance of SignInStore
const signInStore = new SignInStore();

// Register a dispatcher and bind it to handleActions method
AppDispatcher.register(signInStore.handleActions.bind(signInStore));

// export an instance of SignInStore
export default signInStore;
