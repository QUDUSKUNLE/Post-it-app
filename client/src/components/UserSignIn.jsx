import React from 'react';
import GoogleButton from 'react-google-button';
import PropTypes from 'prop-types';
import toastr from 'toastr';
import { Link } from 'react-router-dom';
import firebase from '../vendors/index';
import SignInStore from '../stores/SignInStore';
import { signInAction, signInWithGoogle } from '../actions/signInAction';

/**
 * @description - renders UserSignIn Component
 * @class UserSignIn
 * @extends {React.Component}
 */
export default class UserSignIn extends React.Component {
  /**
   * Create a constructor
   * @constructor
   * @param {object} props -
   */
  constructor(props) {
    super(props);
    this.state = {
      userName: '',
      userId: '',
      email: '',
      password: '',
      loggedIn: false,
      isLoading: false,
      signInResponse: ''
    };

    /**
     * @description This binding is necessary to make `this` work
     * in the callback
     */
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.handleSignInAction = this.handleSignInAction.bind(this);
    this.googleSignIn = this.googleSignIn.bind(this);
    this.handleGoogleEvent = this.handleGoogleEvent.bind(this);
  }

   /**
   * @method componentDidMount
   * @description Adds an event Listener to the Store and fires
   * when the component is fully mounted.
   * @return { void} void
   * @memberof UserSignIn
  */
  componentDidMount() {
    SignInStore.on('SIGN_IN_SUCCESS', this.handleSignInAction);
    SignInStore.on('GOOGLE_SIGN_IN_SUCCESS', this.handleGoogleEvent);
  }

  /**
   * @method componentWillUnmount
   * @description remove event Listener from the Store and fires.
   * @return {void} void
   * @memberof UserSignIn
   */
  componentWillUnmount() {
    SignInStore.removeListener('SIGN_IN_SUCCESS', this.handleSignInAction);
    SignInStore.removeListener('GOOGLE_SIGN_IN_SUCCESS',
      this.handleGoogleEvent);
  }

  /**
   * onChange event
   * @param {object} event - event
   * @return {void} updated state of user
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
   * @description - this handles SignIn form submission
   * @param {object} event - event.
   * @returns {void} .
   */
  onSubmit(event) {
    event.preventDefault();
    signInAction({ ...this.state });
  }

  /**
   * @description This handles handleSignInAction
   * @param {object} user .
   * @returns {void} .
   */
  handleSignInAction() {
    const response = SignInStore.signInUser();
    this.setState({
      loggedIn: true,
    });
    toastr.success(response.message);
    localStorage.setItem('userIn', JSON.stringify(this.state.loggedIn));
    this.props.history.push('/broadcastboard');
  }

  /**
   * @description - this handles Google SignIn Method
   * @param {object} event - event.
   * @returns {void} .
  */
  googleSignIn() {
    const provider = new firebase.auth.GoogleAuthProvider();
    provider.addScope('https://www.googleapis.com/auth/plus.login');
    firebase.auth().signInWithPopup(provider)
      .then((result) => {
        signInWithGoogle(result);
        this.setState({
          isLoading: true
        });
      }).catch(error => {
        toastr.error(error.message);
      });
  }

  /**
   * @memberof UserSignIn
   * @return {*} void
   */
  handleGoogleEvent() {
    const googleResponse = SignInStore.googleSignIn();
    this.setState({
      loggedIn: true,
    });
    toastr.success(googleResponse.message);
    localStorage.setItem('userIn', JSON.stringify(this.state.loggedIn));
    this.props.history.push('/broadcastboard');
  }

  /**
   * @description - render method, React lifecycle method
   * @returns {Object} SignIn component
   * @SignIn
   */
  render() {
    const isLoading = () => {
      const loading = (
        this.state.isLoading ? <div id="loader"></div> : <span></span>
      );
      return loading;
    };
    return (
      <div className="container signin">
        {isLoading()}
        <div className="row">
          <div className="col-md-6 col-md-offset-3">
            <h5 className="text-center">
              <b>Sign in to PostIt</b>
            </h5>
            <form onSubmit={this.onSubmit}
              className="w3-card w3-white" id="signinForm">
              <div className="form-group">
                <label htmlFor="email">Email</label>
                <input value={this.state.email} onChange={this.onChange}
                  id="email" type="email"
                  className="signinform" placeholder="johndoe@example.com"
                  name="email" required />
              </div>
              <div className="form-group">
                <div className="row">
                  <div className="col-md-4">
                    <label htmlFor="password">Password</label>
                  </div>
                </div>
                <input id="password" type="password"
                  value={this.state.password} onChange={this.onChange}
                  className="signinform" placeholder="*********"
                  name="password" required />
              </div>
              <button id="submitButton"
                type="submit" className="signinformbtn">Sign in
              </button>
            </form>
            <br/>
            <span className="pull-right create">
              <Link to="/passwordreset">
                Forgot password?
              </Link></span>
            <br/>
            <center>
              <GoogleButton onClick={this.googleSignIn} />
            </center>
            <br/>
            <div>
              <center>
                <p>New to PostIt? <Link to="/" className="create">
                  Create an account.</Link>
                </p>
              </center>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

// props validation
UserSignIn.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  })
};
