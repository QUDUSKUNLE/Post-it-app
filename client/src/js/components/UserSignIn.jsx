import React from 'react';
import GoogleButton from 'react-google-button';
import PropTypes from 'prop-types';
import firebase from 'firebase';
import { Link } from 'react-router-dom';
import config from '../vendors/vendors.js';
import toastr from 'toastr';
import Footer from './Footer';
import SignInStore from '../stores/SignInStore';
import { signinAction, signInWithGoogle } from '../actions/signInActions.js';
// import { getAllUsers } from '../utils/utils.js';


/**
 * @description - renders SignIn Component
 * @class SignIn
 * @extends {React.Component}
 */
export default class UserSignIn extends React.Component {
  /**
   * Create a constructor
   * @constructor
   * @param {object} props -
   */
  constructor(props) {
    super(props);
    this.state = {
      userName: '',
      userId: '',
      email: '',
      password: '',
      signinMessage: '',
      errMessage: '',
      loggedIn: false
    };
    // Bind input field values
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.handleSignInAction = this.handleSignInAction.bind(this);
    this.googleSignIn = this.googleSignIn.bind(this);
  }

  componentDidMount() {
    SignInStore.on('SIGN_IN_SUCCESS', this.handleSignInAction);
    SignInStore.on('SIGN_IN_ERROR', this.handleSignInAction);
  }

  compoenentWillUnmount() {
    SignInStore.removeListener('SIGN_IN_SUCCESS', this.handleSignInAction);
    SignInStore.removeListener('SIGN_IN_ERROR', this.handleSignInAction);
  }
  /**
   * onChange event
   * @param {object} event - event
   * @return {void} updated state of user
  */
  onChange(event) {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  /**
   * @description - this handles SignIn form submission
   * @param {object} event - event.
   * @returns {void} .
  */
  onSubmit(event) {
    event.preventDefault();
    const user = {
      email: this.state.email,
      password: this.state.password
    };
    signinAction(user);
  }

  /**
   * @description This handles handleSignInAction
   * @param {object} user .
   * @returns {void} .
   */
  handleSignInAction() {
    const response = SignInStore.signInUser();
    if (response.message === 'User Signed in successfully') {
      toastr.success('User Signed in successfully');
      this.setState({
        userName: (Object.values((response.response)[0])[0].userName),
        loggedIn: true,
        email: this.state.email,
        userId: (Object.values((response.response)[0]))[0].userId
      });
      localStorage.setItem('userName', JSON.stringify(this.state.userName));
      localStorage.setItem('userIn', JSON.stringify(this.state.loggedIn));
      localStorage.setItem('Id', JSON.stringify(this.state.userId));
      localStorage.setItem('email', JSON.stringify(this.state.email));
      this.props.history.push('/broadcastboard');
    } else if (response.data.error.code === 'auth/wrong-password') {
      toastr.error(response.data.error.message);
    } else if (response.data.error.code === 'auth/user-not-found') {
      toastr.error(response.data.error.message);
    } else if (response.data.error.code === 'auth/invalid-email') {
      toastr.error(response.data.error.message);
    }
  }


  /**
   * @description - this handles Google SignIn Method
   * @param {object} event - event.
   * @returns {void} .
  */
  googleSignIn() {
    firebase.initializeApp(config);
    const provider = new firebase.auth.GoogleAuthProvider();
    provider.addScope('profile');
    provider.addScope('email');
    firebase.auth().signInWithPopup(provider)
      .then((result) => {
        signInWithGoogle(result)
          .then((res) => {
            this.setState({
              signinMessage: res.data.message,
              userName: res.data.user.displayName,
              loggedIn: true
            });
            toastr.success(this.state.signinMessage);
            localStorage.setItem('userName', JSON.stringify(this.state.userName));
            localStorage.setItem('userIn', JSON.stringify(this.state.loggedIn));
            this.props.history.push('/broadcastboard');
          });
      });
  }

  /**
   * @description - render method, React lifecycle method
   * @returns {Object} SignIn component
   * @SignIn
   */
  render() {
    return (
      <div>
        <nav className="navbar navbar-inverse navabar-fixed-top"
          role="navigation">
          <div className="container-fluid">
            <div className="navbar-header">
              <button type="button" className="navbar-toggle collapsed"
                data-toggle="collapse" data-target=".navbar-collapse">
                <span className="sr-only">Toggle navigation</span>
                <span className="icon-bar"></span>
                <span className="icon-bar"></span>
                <span className="icon-bar"></span>
              </button><Link className="navbar-brand" to="#">
                PostIt<small>App</small>
              </Link></div>
            <div className="collapse navbar-collapse">
              <ul className="nav navbar-nav navbar-right">
                <li><Link to="/">Home</Link></li>
                <li className="active"><Link to="/signin">Sign in</Link></li>
              </ul>
            </div>
          </div>
        </nav>
        <div className="container signin">
          <div className="row">
            <div className="col-md-6 col-md-offset-3">
              <div className="row">
                <center>
                  <GoogleButton onClick={this.googleSignIn}/>
                </center>
                <br/>
                <div className="text-center or"><b>OR</b></div>
                <form onSubmit={this.onSubmit}
                  className="w3-card w3-white" id="signinForm">
                  <div className="form-group">
                    <label htmlFor="email">Email</label>
                    <input value={this.state.email} onChange={this.onChange}
                      id="email" type="email"
                      className="signinform" placeholder="johndoe@example.com"
                      name="email" required />
                  </div>
                  <div className="form-group">
                    <div className="row">
                      <div className="col-md-4">
                        <label htmlFor="password">Password</label>
                      </div>
                      <div className="col-md-4 col-md-offset-4">
                        <Link to="/passwordreset">
                          <h6 className="pull-right create">
                            <b>Forgot password?</b></h6>
                        </Link>
                      </div>
                    </div>
                    <input id="password" type="password"
                      value={this.state.password} onChange={this.onChange}
                      className="signinform" placeholder="*********"
                      name="password" required />
                  </div>
                  <button type="submit" className="signinformbtn">Sign in
                  </button>
                </form>
              </div>
              <br/>
              <div>
                <center>
                  <p>New to PostIt App? <Link to="/" className="create">
								Create an account.</Link>
                  </p>
                </center>
              </div>
            </div>
          </div>
        </div>
        <Footer/>
      </div>
    );
  }
}

// props validation
UserSignIn.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  })
};
