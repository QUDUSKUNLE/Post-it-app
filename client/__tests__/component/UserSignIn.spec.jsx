import React from 'react';
import PropTypes from 'prop-types';
import expect from 'expect';
import sinon from 'sinon';
import { mount } from 'enzyme';
import 'babel-polyfill';
import localStorageMock from '../../src/__mock__/localStorage';
import UserSignIn from '../../src/components/UserSignIn';
import SignInStore from '../../src/stores/SignInStore';
import signInResponse from '../../src/__mock__/signInResponse.json';

window.localStorage = localStorageMock;
describe('<UserSignIn/>', () => {
  const signInAction = sinon.spy();
  sinon.spy(UserSignIn.prototype, 'onSubmit');
  sinon.spy(UserSignIn.prototype, 'componentDidMount');
  const spy = sinon.spy(UserSignIn.prototype, 'componentWillUnmount');

  const mockOnSignIn = sinon.stub(SignInStore,
    'on').callsFake((user, cb) => cb());
  const mockUnMountSignIn = sinon.stub(SignInStore,
    'removeListener').callsFake((user, cb) => cb());
  const mockSignInResponse = sinon.stub(SignInStore,
    'signInUser').returns(signInResponse);

  const props = {
    userName: '',
    userId: '',
    email: '',
    password: '',
    loggeddIn: false,
    isLoading: false,
    history: {
      push: jest.fn()
    },
    signInAction
  };
  const wrapper = mount(<UserSignIn {...props} />,
    {
      childContextTypes: { router: PropTypes.object },
      context: {
        router: {
          history: {
            push: () => '/signin',
            replace: () => null,
            createHref: () => null,
            path: '/signin',
            component: '[function UserSignIn]',
            location: {
              pathname: '/signin',
              search: '',
              hash: '',
              key: 'on2bj3'
            },
            computedMatch: {
              path: '/signin',
              url: '/signin',
              isExact: true,
              params: {}
            }
          }
        }
      }
    }
  );
  it('component to be defined', () => {
    expect(UserSignIn).toBeDefined();
  });
  it('calls componentDidMount', () => {
    expect(UserSignIn.prototype.componentDidMount.calledOnce).toEqual(true);
    UserSignIn.prototype.componentDidMount.restore();
  });
  it('should have signin form fields', () => {
    expect(wrapper.find('[type="email"]').at(0).length).toEqual(1);
    expect(wrapper.find('[type="password"]').at(0).length).toEqual(1);
  });
  it('should called onSubmit method when submit button is clicked', () => {
    wrapper.find('form').simulate('submit');
    expect(UserSignIn.prototype.onSubmit.calledOnce).toEqual(true);
  });
  it('should call onChange method when type something in the input', () => {
    const event = { target: { name: 'name', value: 'value' } };
    wrapper.instance().onChange(event);
    expect(wrapper.state().name).toEqual('value');
  });
  it('expects componentWillUnmount to be unmounted', () => {
    wrapper.instance().componentWillUnmount();
    expect(spy.calledOnce).toBeTruthy();
  });
});

