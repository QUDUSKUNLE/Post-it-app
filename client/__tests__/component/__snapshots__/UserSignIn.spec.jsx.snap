// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserSignIn/> component should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="container signin"
  >
    <span />
    <div
      className="row"
    >
      <div
        className="col-md-6 col-md-offset-3"
      >
        <h5
          className="text-center"
        >
          <b>
            Sign in to PostIt
          </b>
        </h5>
        <form
          className="w3-card w3-white"
          id="signinForm"
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="email"
            >
              Email
            </label>
            <input
              className="signinform"
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="johndoe@example.com"
              required={true}
              type="email"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col-md-4"
              >
                <label
                  htmlFor="password"
                >
                  Password
                </label>
              </div>
            </div>
            <input
              className="signinform"
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="*********"
              required={true}
              type="password"
              value=""
            />
          </div>
          <button
            className="signinformbtn"
            id="submitButton"
            type="submit"
          >
            Sign in
          </button>
        </form>
        <br />
        <span
          className="pull-right create"
        >
          <Link
            replace={false}
            to="/passwordreset"
          >
            Forgot password?
          </Link>
        </span>
        <br />
        <center>
          <GoogleButton
            disabled={false}
            label="Sign in with Google"
            onClick={[Function]}
            type="dark"
          />
        </center>
        <br />
        <div>
          <center>
            <p>
              New to PostIt? 
              <Link
                className="create"
                replace={false}
                to="/"
              >
                Create an account.
              </Link>
            </p>
          </center>
        </div>
      </div>
    </div>
  </div>,
  "nodes": Array [
    <div
      className="container signin"
    >
      <span />
      <div
        className="row"
      >
        <div
          className="col-md-6 col-md-offset-3"
        >
          <h5
            className="text-center"
          >
            <b>
              Sign in to PostIt
            </b>
          </h5>
          <form
            className="w3-card w3-white"
            id="signinForm"
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="email"
              >
                Email
              </label>
              <input
                className="signinform"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="johndoe@example.com"
                required={true}
                type="email"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <div
                className="row"
              >
                <div
                  className="col-md-4"
                >
                  <label
                    htmlFor="password"
                  >
                    Password
                  </label>
                </div>
              </div>
              <input
                className="signinform"
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="*********"
                required={true}
                type="password"
                value=""
              />
            </div>
            <button
              className="signinformbtn"
              id="submitButton"
              type="submit"
            >
              Sign in
            </button>
          </form>
          <br />
          <span
            className="pull-right create"
          >
            <Link
              replace={false}
              to="/passwordreset"
            >
              Forgot password?
            </Link>
          </span>
          <br />
          <center>
            <GoogleButton
              disabled={false}
              label="Sign in with Google"
              onClick={[Function]}
              type="dark"
            />
          </center>
          <br />
          <div>
            <center>
              <p>
                New to PostIt? 
                <Link
                  className="create"
                  replace={false}
                  to="/"
                >
                  Create an account.
                </Link>
              </p>
            </center>
          </div>
        </div>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <UserSignIn
        email=""
        isLoading={false}
        loggeddIn={false}
        password=""
        userId=""
        userName=""
      />,
      "_debugID": 75,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": UserSignIn {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "googleSignIn": [Function],
        "handleGoogleEvent": [Function],
        "handleSignInAction": [Function],
        "onChange": [Function],
        "onSubmit": [Function],
        "props": Object {
          "email": "",
          "isLoading": false,
          "loggeddIn": false,
          "password": "",
          "userId": "",
          "userName": "",
        },
        "refs": Object {},
        "state": Object {
          "email": "",
          "isLoading": false,
          "loggedIn": false,
          "password": "",
          "userId": "",
          "userName": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="container signin"
        >
          <span />
          <div
            className="row"
          >
            <div
              className="col-md-6 col-md-offset-3"
            >
              <h5
                className="text-center"
              >
                <b>
                  Sign in to PostIt
                </b>
              </h5>
              <form
                className="w3-card w3-white"
                id="signinForm"
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="email"
                  >
                    Email
                  </label>
                  <input
                    className="signinform"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="johndoe@example.com"
                    required={true}
                    type="email"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-md-4"
                    >
                      <label
                        htmlFor="password"
                      >
                        Password
                      </label>
                    </div>
                  </div>
                  <input
                    className="signinform"
                    id="password"
                    name="password"
                    onChange={[Function]}
                    placeholder="*********"
                    required={true}
                    type="password"
                    value=""
                  />
                </div>
                <button
                  className="signinformbtn"
                  id="submitButton"
                  type="submit"
                >
                  Sign in
                </button>
              </form>
              <br />
              <span
                className="pull-right create"
              >
                <Link
                  replace={false}
                  to="/passwordreset"
                >
                  Forgot password?
                </Link>
              </span>
              <br />
              <center>
                <GoogleButton
                  disabled={false}
                  label="Sign in with Google"
                  onClick={[Function]}
                  type="dark"
                />
              </center>
              <br />
              <div>
                <center>
                  <p>
                    New to PostIt? 
                    <Link
                      className="create"
                      replace={false}
                      to="/"
                    >
                      Create an account.
                    </Link>
                  </p>
                </center>
              </div>
            </div>
          </div>
        </div>,
        "_debugID": 76,
        "_renderedOutput": <div
          className="container signin"
        >
          <span />
          <div
            className="row"
          >
            <div
              className="col-md-6 col-md-offset-3"
            >
              <h5
                className="text-center"
              >
                <b>
                  Sign in to PostIt
                </b>
              </h5>
              <form
                className="w3-card w3-white"
                id="signinForm"
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="email"
                  >
                    Email
                  </label>
                  <input
                    className="signinform"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="johndoe@example.com"
                    required={true}
                    type="email"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-md-4"
                    >
                      <label
                        htmlFor="password"
                      >
                        Password
                      </label>
                    </div>
                  </div>
                  <input
                    className="signinform"
                    id="password"
                    name="password"
                    onChange={[Function]}
                    placeholder="*********"
                    required={true}
                    type="password"
                    value=""
                  />
                </div>
                <button
                  className="signinformbtn"
                  id="submitButton"
                  type="submit"
                >
                  Sign in
                </button>
              </form>
              <br />
              <span
                className="pull-right create"
              >
                <Link
                  replace={false}
                  to="/passwordreset"
                >
                  Forgot password?
                </Link>
              </span>
              <br />
              <center>
                <GoogleButton
                  disabled={false}
                  label="Sign in with Google"
                  onClick={[Function]}
                  type="dark"
                />
              </center>
              <br />
              <div>
                <center>
                  <p>
                    New to PostIt? 
                    <Link
                      className="create"
                      replace={false}
                      to="/"
                    >
                      Create an account.
                    </Link>
                  </p>
                </center>
              </div>
            </div>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <UserSignIn
    email=""
    isLoading={false}
    loggeddIn={false}
    password=""
    userId=""
    userName=""
  />,
}
`;
